compiler/rustc_typeck/src/check/op.rs

```diff
@@ -429,24 +380,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         )
                         .is_ok()
                     {
-                        if let Ok(lstring) = source_map.span_to_snippet(lhs_expr.span) {
-                            let msg = &format!(
-                                "`{}{}` can be used on `{}`, you can dereference `{}`",
-                                op.node.as_str(),
-                                match is_assign {
-                                    IsAssign::Yes => "=",
-                                    IsAssign::No => "",
-                                },
-                                lhs_deref_ty.peel_refs(),
```

Comment:
Peeling the type here is actually wrong -- when we have `/* &&i32 */ + /* i32 */` and we deref the LHS, we are adding `&i32` and `i32`, not `i32` and `i32`.